docs/loading.md

```diff
@@ -0,0 +1,104 @@
+# Loading and navigation
+
+Playwright logically splits the process of showing a new document in the page into **navigation** and **loading**.
+
+**Navigation** starts with an intent, for example:
+- loading a url with `page.goto('http://example.com')`;
+- clicking a link with `page.click('text="Continue"')`;
+- reloading with `page.reload()`;
+- assiging to location in the script like `location.href = 'http://example.com'`;
+- using [history api](https://developer.mozilla.org/en-US/docs/Web/API/History) in the script like `history.pushState({}, 'title', '#deep-link')`.
+
+This navigation intent may result in a navigation or get canceled, for example transformed into a download. When navigation succeeds, page starts **loading** the document.
+
+**Loading** usually takes time, retrieving the response over the network, parsing, executing scripts and firing events. The typical loading scenario is:
+- `page.url()` is set to the loading url;
+- document content is loaded over network and parsed;
+- `domcontentloaded` event is fired;
+- page executes some scripts and loads resources like stylesheets and images;
+- `networkidle2` event is fired;
```

Comment:
Should we add a short description of the `networkidle2` and `networkidle0` events here?